src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -430,54 +448,70 @@ partial class BoundLiteral : ILiteralExpression
 
     partial class BoundObjectCreationExpression : IObjectCreationExpression
     {
-        private static readonly ConditionalWeakTable<BoundObjectCreationExpression, object> s_memberInitializersMappings =
-            new ConditionalWeakTable<BoundObjectCreationExpression, object>();
+        private ComputedState _lazyComputedState;
+
+        private ComputedState GetComputedState()
+        {
+            if (_lazyComputedState == null)
+            {
+                var constructorArguments = BoundCall.DeriveArguments(this.Arguments, this.ArgumentNamesOpt, this.ArgsToParamsOpt, this.ArgumentRefKindsOpt, this.Constructor.Parameters);
+                var memberInitializers = GetMemberInitializers(this.InitializerExpressionOpt as BoundObjectInitializerExpression);
+                Interlocked.CompareExchange(ref _lazyComputedState, new ComputedState(constructorArguments, memberInitializers), null);
```

Comment:
I think you can just assign here. You are not relying on uniqueness of the ComputedState instance.